scripts/yocto-build-and-test.sh

```diff
@@ -478,58 +474,72 @@ build_and_test_client() {
 
         cd $BUILDDIR
 
-        # Base image clean
+        # Base image
+        local images_to_build=$image_name
+
+        if is_building_extra_images_for_board "$board_name"; then
+            if [[ $image_name == core-image-full-cmdline ]]; then
+                images_to_build+=" mender-image-full-cmdline-rofs"
+            fi
+            if has_component monitor-client \
+                    && [[ $image_name == core-image-full-cmdline ]] \
+                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-monitor-image-full-cmdline.bb ]]; then
+                images_to_build+=" mender-monitor-image-full-cmdline"
+            fi
+            if has_component mender-gateway \
+                    && [[ $image_name == core-image-full-cmdline ]] \
+                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-gateway-image-full-cmdline.bb ]]; then
+                images_to_build+=" mender-gateway-image-full-cmdline"
+            fi
+        fi
```

Comment:
This is correct. An slightly more maintainable way could be:
```suggestion
        if is_building_extra_images_for_board "$board_name" \
                    && [[ $image_name == core-image-full-cmdline ]]; then
            images_to_build+=" mender-image-full-cmdline-rofs"
            if has_component monitor-client \
                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-monitor-image-full-cmdline.bb ]]; then
                images_to_build+=" mender-monitor-image-full-cmdline"
            fi
            if has_component mender-gateway \
                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-gateway-image-full-cmdline.bb ]]; then
                images_to_build+=" mender-gateway-image-full-cmdline"
            fi
        fi
```

Reply:
Noted! I merged to get the speed improvement now, and I will submit this separately.